src/node_alive/server.py

```diff
@@ -0,0 +1,130 @@
+import rospy
+import rosnode
+from diagnostic_msgs.msg import DiagnosticArray, DiagnosticStatus, KeyValue
+from std_msgs.msg import String
+from node_alive.srv import ListNodesAlive, ListNodesAliveResponse
+import fnmatch
+from enum import Enum
+from rosnode import rosnode_ping_all
+
+
+class NodeState(Enum):
+    """
+    Enum for representing the node state with a descriptive name
+    """
+    Alive = str(0)
```

Comment:
* Why not use ALLCAPS (I believe this is common for enums)?
* The values (0, 1, 2, 3) are not very descriptive. Since you're sending strings anyway, is there a specific reason for these values (in those cases it's common to encode that in the message itself)

Reply:
It has to be strings as in KeyValuePair the value has to be a string. This is merged code. It has always been only one single capital letter. I don't mind changing it to all caps.

Reply:
* I'd change it to all-caps then (boyscout rule ;-). 
* With my second comment, I was wondering why we don't use
```python
...
   ALIVE = "alive" 
...
```
This would be much clearer for the recipient including inspections (e.g. rostopic echo or rqt console). Therefore: would it break anything if we did?

Reply:
We might be following the `diagnostics_msgs`, http://docs.ros.org/en/api/diagnostic_msgs/html/msg/DiagnosticStatus.html, here, I am not sure though. Maybe @reinzor knows

Reply:
These are just arbitrary key value pairs. Don't think anybody is interpreting this value. I agree with @jlunenburg ; also conform https://docs.python.org/3/library/enum.html

Reply:
That could also work.